crates/transport_iroh/src/lib.rs

```diff
@@ -373,6 +355,215 @@ impl IrohTransport {
     }
 }
 
+/// Spawns an asynchronous task to continuously read and handle incoming uni-directional
+/// streams from an iroh connection. There is only one stream at a time incoming from a
+/// remote. It's persisted in memory until the connection is closed or an error occurs.
+///
+/// Errors when receiving the preflight frame lead to a break of the loop accepting
+/// incoming streams. The preflight must succeed for data frames to be accepted. The
+/// connection cannot recover from a failed preflight, and a new connection must be
+/// established.
+///
+/// # Parameters
+/// - `ctx`: The connection context containing handler and remote URL state.
+/// - `connections`: Shared map of peer URLs to their connection contexts, updated when
+///   the preflight succeeds.
+/// - `local_url`: The local URL for this endpoint, used to respond to preflight messages.
+fn spawn_connection_reader(
+    ctx: Arc<ConnectionContext>,
+    connections: Connections,
+    local_url: Arc<RwLock<Option<Url>>>,
+) {
```

Comment:
No handle returned? Seems like something we should be able to stop as part of closing down a space?

Reply:
Good catch! Fixed https://github.com/holochain/kitsune2/pull/411/changes/9b87c26911c1f9f591855ab56daa84dcc4d5f41e